src/controller/CHIPDeviceController.cpp

```diff
@@ -118,14 +118,24 @@ CHIP_ERROR DeviceController::Init(ControllerInitParams params)
     mDNSResolver.SetCommissioningDelegate(this);
     RegisterDeviceDiscoveryDelegate(params.deviceDiscoveryDelegate);
 
-    VerifyOrReturnError(params.operationalCredentialsDelegate != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
-    mOperationalCredentialsDelegate = params.operationalCredentialsDelegate;
-
     mVendorId = params.controllerVendorId;
     if (params.operationalKeypair != nullptr || !params.controllerNOC.empty() || !params.controllerRCAC.empty())
     {
         ReturnErrorOnFailure(InitControllerNOCChain(params));
     }
+    else if (params.storageFabricIndex.HasValue())
+    {
+        VerifyOrReturnError(params.systemState->Fabrics()->FabricCount() > 0, CHIP_ERROR_INVALID_ARGUMENT);
+        if (params.systemState->Fabrics()->FindFabricWithIndex(params.storageFabricIndex.Value()) != nullptr)
+        {
+            mFabricIndex = params.storageFabricIndex.Value();
```

Comment:
Where is the checking that there is no other controller with that mFabricIndex?  I know InitControllerNOCChain doesn't have that either, apparently, but it really needs to happen, especially if the fabric index is being provided externally.  Having two controllers with the same mFabricIndex will lead to nasty security bugs...

Reply:
It may be difficult to check for this on existing basis. Have left a TODO for this.